synapse/app/homeserver.py

```diff
@@ -254,6 +254,17 @@ def run_startup_checks(self, db_conn, database_engine):
         except IncorrectDatabaseSetup as e:
             quit_with_error(e.message)
 
+    def get_db_conn(self):
+        db_conn = self.database_engine.module.connect(
+            **{
+                k: v for k, v in self.db_config.get("args", {}).items()
+                if not k.startswith("cp_")
```

Comment:
It might be nice to add comment to explain that this is stripping the connection pool config.